src/integration/test_suites/smoke_test_fc_secret_partition_zeroization/smoke_test_fc_secret_partition_zeroization.c

```diff
@@ -0,0 +1,168 @@
+// limitations under the License.
+//********************************************************************************
+#include <string.h>
+#include <stdint.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include "soc_address_map.h"
+#include "printf.h"
+#include "riscv_hw_if.h"
+#include "soc_ifc.h"
+#include "caliptra_ss_lc_ctrl_address_map.h"
+#include "caliptra_ss_lib.h"
+#include "fuse_ctrl.h"
+#include "lc_ctrl.h"
+#include "fuse_ctrl_mmap.h"
+
+volatile char* stdout = (char *)SOC_MCI_TOP_MCI_REG_DEBUG_OUT;
+#ifdef CPT_VERBOSITY
+    enum printf_verbosity verbosity_g = CPT_VERBOSITY;
+#else
+    enum printf_verbosity verbosity_g = LOW;
+#endif
+
+/**
+ * Program a LCC transition token in `VENDOR_SECRET_PROD_PARTITION` zeroize the secret
+ * partition and verify that the valid bit is asserted once the partition has been locked.
+ * The test proceeds in the following steps:
+ *
+ *   1. Write a full transition token into the partition.
+ *   2. Read back the token and verify it is equal to the token written in Step 1.
+ *      This works as long as the secret partition has not been locked.
+ *   3. Verify that the partition's digest register is 0.
+ *   4. Lock the partition by calculating a hardware digest.
+ *   5. Reset the RTL.
+ *   6. Try to read the token and verify that it results in an error as the
+ *      partition is secret and locked, thus not accessible by software anymore.
+ *   7. Try to write a value into the partition and verify that it results in an error
+ *      as the partition has been locked in Step 5.
+ *   8. Read back the digest from the partition's digest register and verify it
+ *      is non-zero.
+ *   9. Apply zeroization
+ *   10.Verify the valid signal drops via SVA.
+ */
+
+
+void secret_partition_zeroization(void) {
+  const uint32_t base_address = partitions[VENDOR_SECRET_PROD_PARTITION].address;
+  const uint32_t fuse_address = CPTRA_SS_TEST_EXIT_TO_MANUF_TOKEN;
+
+  const uint32_t data[4] = {0xdeadbeef, 0xcafebabe, 0x12345678, 0xabababab};
+  uint32_t read_data[4];
+  uint32_t zero_r_data[2];
+  uint32_t zeroize_num_times = 3;
+
+  // Step 1. Write a full transition token into the partition.
+  dai_wr(fuse_address, data[0], data[1], 64, 0);
+  dai_wr(fuse_address+8, data[2], data[3], 64, 0);
+
+  // Step 2. Read back the token and verify it is equal to the token written in Step 1.
+  //         This works as long as the secret partition has not been locked.
+  dai_rd(fuse_address, &read_data[0], &read_data[1], 64, 0);
+  dai_rd(fuse_address+8, &read_data[2], &read_data[3], 64, 0);
+  for (uint32_t idx = 0; idx < 4; idx++) {
+    if (data[idx] != read_data[idx]) {
+      VPRINTF(LOW, "ERROR: incorrect fuse data[index=%0d]: expected: %08X actual: %08X\n", idx, data[idx], read_data[idx]);
+      goto epilogue;
+    }
+  }
+  // Step 3. Verify that the partition's digest register is 0.
+  uint32_t digest[2];
+  digest[0] = lsu_read_32(OTP_CTRL_SECRET_PROD_PARTITION_0_DIGEST_DIGEST_0);
```

Comment:
I thought the digest registers were populated after reset. Is it possible to actually read them before reset?